### ISSUE TITLE
WFS Catalog do not work with mapserver or if have parameters

### ISSUE THREAD

**Issue by offtherailz** [geosolutions-it/MapStore2#8940]
## Description

If you add one or more parameters to a WFS catalog, mapstore do not add the standard parameters to the WFS getCapabilities (service, version, request). 
We can add them all manually in the URL, but then the WFS requests will include the request=getCapabilities.

Modifying the layers in map JSON like this:

```json
 {
                "id": "temperature__fca1e850-9e68-11ed-95a1-058253ffdf19",
                "search": {
                    "url": "http://h2901511.stratoserver.net/cgi-bin/mapserv?map=/var/www/html/transfer/mapfile.map",
                    "type": "wfs"
                },
                "name": "temperature",
                "description": "",
                "style": {
                    "weight": 1,
                    "color": "rgba(0, 0, 255, 1)",
                    "opacity": 1,
                    "fillColor": "rgba(0, 0, 255, 0.1)",
                    "fillOpacity": 0.1,
                    "radius": 10
                },
                "title": "temperature",
                "type": "wfs",
                "url": "http://h2901511.stratoserver.net/cgi-bin/mapserv?map=/var/www/html/transfer/mapfile.map",
                "bbox": {
                    "bounds": {
                        "minx": 9.80166610784991,
                        "miny": 52.4075800424749,
                        "maxx": 13.3900734292828,
                        "maxy": 52.5199710468719
                    },
                    "crs": "EPSG:4326"
                },
                "visibility": true,
                "singleTile": false,
                "dimensions": [],
                "hideLoading": false,
                "handleClickOnLayer": false,
                "useForElevation": false,
                "hidden": false
            }
```
anyway the layer have an error due to not loading DescribeFeatureType (this should be mandatory?). 

## How to reproduce

1. 
-  Open the catalog tool
- add a new catalog
- Select WFS
- insert this URL `https://gs-stable.geo-solutions.it/geoserver/wfs?test_1=parameter`

*Expected Result*

The catalog browses the service. You should be able to see the layers.
The requests should be :   `https://gs-stable.geo-solutions.it/geoserver/wfs?test_1=parameter&service=WFS&version=1.1.0&request=GetCapatbilities`

*Current Result*

The catalog do not show any results 
The URL in the request do not contain standard parameters

2.

USING this as URL in catalog: 

 `https://gs-stable.geo-solutions.it/geoserver/wfs?test_1=parameter&service=WFS&version=1.1.0&request=GetCapatbilities`

You can browse and add the layers, but layers fails if describe is not responding and because the service receives the parameters above. 

The request getFeatures do not contain the proper parameters, but Instead the getcapabilities request, so it fails. The catalog should be smart enough to recognize and remove standard parameters

3. 

- Some servers do not support JSON format for describeFeatureType, this blocks mapstore. 
- Same for getFeature requests. 

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

**Comment by jratike80** [comment]
A notice about

> Some servers do not support JSON format for describeFeatureType, this blocks mapstore.
> 
> Same for getFeature requests.

The only format that WFS servers must support by the standard is GML.
Mapserver can be configured so that it does not need &map= parameter (that is also recommended), and Mapserver can be configured to support GeoJSON as outputformat. I doubt that with DescribeFeatureType.

**Comment by offtherailz** [comment]
Yes, for this reason I'd say that MapStore supports only WFS servers that support these JSON formats. This should be clearely indicated on doc (at least while we do support only this format).
